src/pepr/operator/controllers/network/generate.ts

```diff
@@ -0,0 +1,134 @@
+import { V1LabelSelector, V1NetworkPolicyPeer, V1NetworkPolicyPort } from "@kubernetes/client-node";
+import { kind } from "pepr";
+
+import { Allow, RemoteGenerated } from "../../crd";
+import { anywhere } from "./generators/anywhere";
+import { cloudMetadata } from "./generators/cloudMetadata";
+import { intraNamespace } from "./generators/intraNamespace";
+import { generateKubeAPI } from "./generators/kubeAPI";
+
+export function generate(namespace: string, policy: Allow): kind.NetworkPolicy {
+  // Create a unique name for the NetworkPolicy based on the package name, index, direction, pod labels, and port
+  const name = generateName(policy);
```

Comment:
Similar question about uniqueness here, it does not look like `generateName` is scoping the name per package.

Reply:
that is correct, that is handled in the policies.ts file